### DISCUSSION TITLE
Where to find the command IDs for Top Menu Bar in Theia

### DISCUSSION THREAD

**Discussion by sboudreaux96** [eclipse-theia/theia#9577] [Q&A] [answered]
Hi - I know that it is possible to disable and hide specific menu items within the top menu bar items (like "new file", "new folder", etc.), but I cannot find where the command IDs are for the actual menu items themselves (File, Edit, Selection, etc.). Where do we find these? 
![image](https://user-images.githubusercontent.com/83592423/121547193-eb6ff000-c9d9-11eb-973a-32b05be20010.png)

**Comment by vince-fugnitto** [comment, marked as answer]
@sboudreaux96 see which command paths are added to the `MAIN_MENU_BAR` path.

For example, `File` is:

https://github.com/eclipse-theia/theia/blob/64e613fd31bb1adeb8e8078bc9b8725a21f2182b/packages/core/src/browser/common-frontend-contribution.ts#L59

and it is contributed by:

https://github.com/eclipse-theia/theia/blob/64e613fd31bb1adeb8e8078bc9b8725a21f2182b/packages/core/src/browser/common-frontend-contribution.ts#L447-L450

**Reply by sboudreaux96** [reply]
@vince-fugnitto Thanks for the quick reply! So in the case of "file" how would you access that command ID? I am trying to find the ID to specifically disable a whole menu item rather than targeting every item in the menu (when I do that the menu item, e.g. terminal, is still there even though all menu items within it are disabled).

**Reply by vince-fugnitto** [reply]
I believe you can do the following [link](https://spectrum.chat/theia/general/how-to-disable-menu-sub-menu-items-via-my-custom-extensions~306b4799-fe97-4011-887f-f8dd07e4e51d?m=MTU2NDY2Mzg1OTM3Ng==):

```ts
this.menuModelRegistry.unregisterMenuAction(CommonMenus.FILE.slice(-1)[0]);
```

**Reply by sboudreaux96** [reply]
Ok I understand that now - where would I find the location of the "Selection" or "Terminal" Menu items? I am having a hard time finding where I can find those in the code base.

**Reply by vince-fugnitto** [reply]
@sboudreaux96 it is quite easy if you search the codebase based on the `label` or as I mentioned in https://github.com/eclipse-theia/theia/discussions/9577#discussioncomment-853330 looking for references of `MAIN_MENU_BAR`.

A simple search yields the results. 
- https://github.com/eclipse-theia/theia/search?q=MAIN_MENU_BAR

https://github.com/eclipse-theia/theia/blob/7d3b1d3b33cb565ec07965e5baf10c926006cdfd/packages/monaco/src/browser/monaco-menu.ts#L28

https://github.com/eclipse-theia/theia/blob/7d3b1d3b33cb565ec07965e5baf10c926006cdfd/packages/terminal/src/browser/terminal-frontend-contribution.ts#L59

**Reply by sboudreaux96** [reply]
ok thanks
